extension/parquet/parquet_writer.cpp

```diff
@@ -751,8 +752,17 @@ struct DecimalStatsUnifier : public NumericStatsUnifier<T> {
 		if (stats.empty()) {
 			return string();
 		}
-		auto numeric_val = Load<T>(const_data_ptr_cast(stats.data()));
-		return Value::DECIMAL(numeric_val, width, scale).ToString();
+
+		auto stats_data = const_data_ptr_cast(stats.data());
+
+		if (sizeof(T) == sizeof(hugeint_t)) {
+			auto _schema = ParquetColumnSchema();
+			auto numeric_val = ParquetDecimalUtils::ReadDecimalValue<hugeint_t>(stats_data, stats.size(), _schema);
+			return Value::DECIMAL(numeric_val, width, scale).ToString();
+		} else {
+			auto numeric_val = Load<T>(stats_data);
+			return Value::DECIMAL(numeric_val, width, scale).ToString();
```

Comment:
Is the second case still needed? Shouldn't `ParquetDecimalUtils::ReadDecimalValue` cover both cases?

Reply:
Actually, I can see you haven't run CI for this branch yet on your fork. Could you try doing that first and seeing if everything is green? That way, we don't overload CI in this repository.